### PR TITLE
libcdb: add `offline_only` to `search_by_symbol_offsets`

### PR DESCRIPTION
**Pull request by the-soloist**
Split [#2259](https://github.com/Gallopsled/pwntools/pull/2259).

### CODE REVIEW

File: pwnlib/libcdb.py [hunk 1]
```diff
@@ -143,6 +143,39 @@ def provider_local_database(hex_encoded_id, hash_type):
...
+        syms = _parse_libc_symbol(symbol_file)
+
+        matched = 0
+        for name, addr in symbols.items():
+            if isinstance(addr, str):
+                addr = int(addr, 16) 
+
+            # Compare last 12 bits
+            if syms.get(name) and (syms.get(name) & 0xfff) == (addr & 0xfff):
+                matched += 1
```
Comment (peace-maker) [inline pwnlib/libcdb.py:169]: We can speed this up by aborting this loop once there was a mismatch.

File: pwnlib/libcdb.py [hunk 2]
```diff
@@ -592,26 +628,49 @@ def search_by_symbol_offsets(symbols, select_index=None, unstrip=True, return_as
...
+
     if return_as_list:
-        return [libc['buildid'] for libc in matching_libcs]
+        return [libc['buildid'] for libc in matching_list]
 
-    if len(matching_libcs) == 1:
-        return search_by_build_id(matching_libcs[0]['buildid'], unstrip=unstrip)
+    # If there's only one match, return it directly
+    if len(matching_list) == 1:
+        return search_by_build_id(matching_list[0]['buildid'], unstrip=unstrip)
```
Comment (peace-maker) [inline pwnlib/libcdb.py]: Maybe make sure to pass the `offline_only` argument through here and the other `search_by_build_id` calls in this function. I think it aborts when finding it offline before trying online, but just to be sure for the future.

File: pwnlib/libcdb.py [hunk 3]
```diff
@@ -592,26 +628,49 @@ def search_by_symbol_offsets(symbols, select_index=None, unstrip=True, return_as
...
 
+    # If a specific index is provided, validate it and return the selected libc
     if select_index is not None:
-        if select_index > 0 and select_index <= len(matching_libcs):
-            return search_by_build_id(matching_libcs[select_index - 1]['buildid'], unstrip=unstrip)
+        if select_index > 0 and select_index <= len(matching_list):
```
Comment (peace-maker) [inline pwnlib/libcdb.py:670]: Is the `Path.rglob` order stable? Otherwise selecting the result by index might not work consistently here.

**Review by peace-maker — CHANGES_REQUESTED** [review]
Cool, thank you for working on this!

**Comment by the-soloist** [comment]
Thank you for pointing out problems. I have completed the modifications.

**Review by peace-maker — APPROVED** [review]

**Pull request merged**